apps/web/src/app/(profile-v2)/[username]/v2/_components-v2/badges.tsx

```diff
@@ -0,0 +1,82 @@
+import type { BadgeInfo } from '~/app/(profile)/[username]/_components/dashboard/_actions';
+import { SlugToBadgeIcon } from '~/app/(profile)/[username]/_components/dashboard/badges';
+import {
+  Tooltip,
+  TooltipContent,
+  TooltipProvider,
+  TooltipTrigger,
+} from '@repo/ui/components/tooltip';
+
+export function Badges(props: { data: BadgeInfo[] }) {
+  const data = [
+    { slug: 'aot-2023-bronze', name: 'Advent of TypeScript 2023 Bronze' },
+    { slug: 'aot-2023-silver', name: 'Advent of TypeScript 2023 Bronze' },
+    { slug: 'aot-2023-gold', name: 'Advent of TypeScript 2023 Bronze' },
+    { slug: 'aot-2023-platinum', name: 'Advent of TypeScript 2023 Bronze' },
+  ];
+  return (
+    <div>
+      <TooltipProvider>
+        <BadgesStuff data={data} />
+      </TooltipProvider>
+    </div>
+  );
+}
+
+function BadgesStuff(props: { data: BadgeInfo[] }) {
+  return (
+    <div className="grid grid-cols-3 gap-4  ">
+      {props.data.map((b) => {
+        const Icon = SlugToBadgeIcon[b.slug];
+        return (
+          <Tooltip key={b.slug}>
+            <TooltipTrigger>
+              <Icon className="h-16 w-16" />
+            </TooltipTrigger>
+            <TooltipContent>
+              <p>{b.name}</p>
+            </TooltipContent>
+          </Tooltip>
+        );
+      })}
+      {props.data.length < 9
+        ? Array.from({ length: 9 - props.data.length }).map((i) => (
+            <EmptyBadge key={`empty-badge-${i}`} />
+          ))
+        : null}
+    </div>
+  );
+}
+
+function EmptyBadge() {
+  return (
+    <svg
+      width="52"
+      height="60"
+      viewBox="0 0 52 60"
+      fill="none"
+      className="h-16 w-16"
+      xmlns="http://www.w3.org/2000/svg"
+    >
+      <mask
+        id="mask0_408_2530"
+        style={{
+          maskType: 'alpha',
+        }}
+        maskUnits="userSpaceOnUse"
+        x="0"
+        y="0"
+        width="52"
+        height="60"
+      >
+        <path
```

Comment:
I have a similar idea in my PR but I emptied this path as it shows the "2023". I also noticed the badge has a Christmas Tree Outline. 

At any rate, I like the idea of abstracting the empty badges among bronze->silver->gold->platinum so we can just attach the appropriate badge image and give it some label like "SharedSolutions". The details of this is way more up your expertise to implement than mine.